src/httpnotifier.cpp

```diff
@@ -64,12 +74,16 @@ bool HttpNotifier::send_notify(const std::string& impu,
   rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
   notification.Accept(writer);
 
-  std::map<std::string, std::string> headers;
   std::string body = buffer.GetString();
 
-  HTTPCode http_code = _http_connection->send_post(_http_url_path,
-                                                   headers,
-                                                   body,
-                                                   trail);
+  std::unique_ptr<HttpRequest> req(new HttpRequest(_http_connection->_server, 
```

Comment:
No, this is wrong - we designed a shiny new API for a reason.

At a guess, this doesn't compile under `make` because `_http_connection->server` is protected.

This should be:

```
HTTPCode http_code = _http_connection->create_request(HttpClient::RequestType::POST, _http_url_path)
   ->set_body(body)
   .set_sas_trail(trail)
   .send()
   .get_rc();
```